### PR TITLE
Navigation and Suggestion bug fixes

### PR DESCRIPTION
**Pull request by psim3141**

### CODE REVIEW

**Comment by psim3141** [comment]
This fixes:
https://trello.com/c/PNHixYlp
https://trello.com/c/EHSqNBu5

File: app/components/Puzzle/Context.jsx [hunk 1]
```diff
@@ -315,7 +329,14 @@ const PuzzleContextProvider = ({ initialGrid, puzzleId, children }) => {
...
   const prevDownClue = () => {
-    const [row, column] = activeCell;
+    const [activeRow, activeColumn] = activeCell;
+    let row = activeRow;
+    let column = activeColumn;
+    
+    if (grid[activeRow][activeColumn].clue.downClueNumber && !grid[activeRow][activeColumn].clue.isDownStart){
+      [row, column] = findClueStartCell(grid, grid[activeRow][activeColumn].clue.downClueNumber, "down");
+    }     
+
```
Comment (jasonrhodes) [inline app/components/Puzzle/Context.jsx:339]: So @psim3141 these changes are checking if we are in a clue, just go to the start of that clue, vs. going to the previous clue (and vice versa for the next)? Could we do those checks in the "getPrevDownClueStart" and "getNextDownClueStart" functions or will they not work there?

Comment (psim3141) [reply app/components/Puzzle/Context.jsx:339]: I think they would work there as well, should I modify the code?

**Comment by psim3141** [comment]
I moved the functionality around

**Pull request merged**